### ISSUE TITLE
MockedResponse should set generic type to FetchResult

### ISSUE THREAD

**Issue by charesbast** [apollographql/apollo-client#5504]
The `MockedResponse` interface, needed for the `MockedProvider` mocks prop currently doesn't allow us to check the result type. With a small fix it could be done :)

**Intended outcome:**
```
export interface MockedResponse<ResultData> {
    request: GraphQLRequest;
    result?: FetchResult<ResultData> | ResultFunction<FetchResult<ResultData>>;
    error?: Error;
    delay?: number;
    newData?: ResultFunction<FetchResult<ResultData>>;
}
```

**Actual outcome:**
```
export interface MockedResponse {
    request: GraphQLRequest;
    result?: FetchResult | ResultFunction<FetchResult>;
    error?: Error;
    delay?: number;
    newData?: ResultFunction<FetchResult>;
}
```

**How to reproduce the issue:**

**Versions**
@apollo/react-testing: 3.1.3

**Comment by MarcelReis** [comment]
Fixed by #6015

**Comment by hwillson** [comment]
As mentioned in https://github.com/apollographql/apollo-client/issues/5504#issuecomment-756227484, thanks was fixed. Thanks!
